packages/jest-core/src/cli/index.ts

```diff
@@ -167,13 +155,28 @@ const buildContextsAndHasteMaps = async (
   return {contexts, hasteMapInstances};
 };
 
-const _run10000 = async (
+/**
+ * Runs Jest either in watch mode or as a one-off. This is a lower-level API than `runCLI` and is intended for internal use by `runCLI` or externally.
+ * Note that `process.exit` might be called when using `globalConfig.watch` or `globalConfig.watchAll` is true.
+ *
+ * @param globalConfig The global configuration to use for this run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param configs The project configurations to run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param warnForDeprecations Whether or not to warn for deprecation messages when `globalConfig.watch` or `globalConfig.watchAll` is true.
+ * @param outputStream The stream to write output to. If not provided, it defaults to `process.stdout`.
+ * @returns A Promise that resolves to the result, or never resolves when `globalConfig.watch` or `globalConfig.watchAll` is true.
+ * @example
+ * import { runCore, readConfigs } from 'jest';
+ *
+ * const { globalConfig, configs } = await readConfigs(process.argv, [process.cwd()]);
+ * const results = await runCore(globalConfig, configs);
+ * console.log(results);
+ */
+export const runCore = async (
   globalConfig: Config.GlobalConfig,
   configs: Array<Config.ProjectConfig>,
-  hasDeprecationWarnings: boolean,
-  outputStream: NodeJS.WriteStream,
-  onComplete: OnCompleteCallback,
-) => {
+  warnForDeprecations = false,
+  outputStream: NodeJS.WriteStream = process.stdout,
```

Comment:
Perhaps a new function should be added for programmatic usage instead of exposing this one? It looks unclear why `outputStream` must be passed if there is `useStderr` option. The new function could simply wrap this one similar to `runCli` and it could pass the `outputStream`:

https://github.com/facebook/jest/blob/285b40d5954fd81e07b02db92185fd317ba92578/packages/jest-core/src/cli/index.ts#L50-L51

Reply:
Yes, I can do that. That way I can also remove the deprecation warnings from the api.